src/Hocon/Impl/HoconObject.cs

```diff
@@ -29,7 +29,7 @@ namespace Hocon
     /// }
     /// </code>
     /// </summary>
-    public class HoconObject : Dictionary<string, HoconField>, IHoconElement
+    public class HoconObject : SortedDictionary<string, HoconField>, IHoconElement
```

Comment:
Have this object implement `IReadOnlyDictionary` (or equivalent) or use a `SortedDictionary` internally